training/src/main/scala/com/airbnb/aerosolve/training/NDTree.scala

```diff
@@ -100,36 +120,39 @@ object NDTree {
     if (!makeLeaf) {
       val deltas = getDeltas(bounds)
 
-      // Choose axisIndex with largest corresponding delta
-      val axisIndex = deltas.zipWithIndex.maxBy(_._1)._2
-
-      val split = splitType match {
-        case SplitType.Median =>
-          getSplitUsingMedian(points, indices, axisIndex)
-        case SplitType.SurfaceArea =>
-          getSplitUsingSurfaceArea(points, indices, bounds, axisIndex)
-      }
-
-      if (split.leftIndices.length <= options.minLeafCount &&
-          split.rightIndices.length <= options.minLeafCount) {
-        log.debug(s"${split.splitValue} leftIndices ${split.leftIndices.length} ${split.rightIndices.length}")
+      // TODO Choose axisIndex with largest corresponding delta as option: deltas.zipWithIndex.maxBy(_._1)._2
+      val axisIndex:Int = getNextAxis(lastAxis, deltas, nodes(0), options.minLeafWidthPercentage)
```

Comment:
Did you mean to pop the head in the process? as is, would `nodes(0)` always point to the same node in subsequent calls?